netmaster/master/netmaster.go

```diff
@@ -501,17 +501,26 @@ func CreateNetwork(network intent.ConfigNetwork, stateDriver core.StateDriver, t
 		// XXX: do configuration check, to make sure it is allowed
 	}
 
+	subnetIsAllocated := false
 	if nwCfg.SubnetIP == "" {
 		nwCfg.SubnetLen = gCfg.Auto.AllocSubnetLen
 		nwCfg.SubnetIP, err = gCfg.AllocSubnet(rm)
 		if err != nil {
 			return err
 		}
+		subnetIsAllocated = true
 	}
 
 	nwCfg.DefaultGw = network.DefaultGw
-	if nwCfg.DefaultGw == "" {
-		// TBD: allocate per global policy
+	// For auto derived subnets assign gateway ip be the last valid unicast ip the subnet
+	if nwCfg.DefaultGw == "" && subnetIsAllocated {
```

Comment:
I am wondering, how will this work when a container has two interfaces(aka endpoints) in different networks. I think if we derive a default-gw for both networks then this will break.

May be instead of default-gw, we should have gateway per network. And then have a tenant level default-gw config?

Also is it possible to have a few tests for this?